### PR TITLE
Add functionality to domain whois requests.

### PR DESCRIPTION
**Pull request by DesBlock**
Added functionality to:
Properly query root domains ie. google.com instead of mail.google.com
Add a section of output to account for subdomains that were queried
Reduce redundant queries of root domains

### CODE REVIEW

File: cmd/domain.go [hunk 1]
```diff
@@ -71,6 +88,19 @@ func domainScannerLogic(ctx *cobra.Command, readr io.ReadWriteSeeker) []rdap.Who
...
+	// Extract root domain from provided domain.
+	rootDomain, err := publicsuffix.EffectiveTLDPlusOne(domain)
+	if err != nil {
+		return rootDomain, err
+	}
+	// Check to see if root domain is different than domain provided.
+	if rootDomain != domain {
+		cli.Debug("Using Root Domain '%v' instead of '%v'", rootDomain, domain)
+	}
+	return rootDomain, err
```
Comment (brutalgg) [inline cmd/domain.go]: The valid return should return rootDomain, nil

File: cmd/domain.go [hunk 2]
```diff
@@ -57,11 +59,26 @@ func domainScannerLogic(ctx *cobra.Command, readr io.ReadWriteSeeker) []rdap.Who
...
+			// Check if root romain has been queried
+			for index, entry := range result {
+				if entry.Name == strings.ToUpper(rootDomain) {
+					//Root domain has already been queried, add to root domain entry "DomainsSearched" .
+					result[index].DomainsSearched = append(result[index].DomainsSearched, strings.ToUpper(i))
+					//Skip query and exit out of current iteration of ScannerLoop.
+					continue ScannerLoop
+				}
+			}
+			if r, e := queryDomain(rootDomain); e != nil {
```
Comment (brutalgg) [inline cmd/domain.go]: The first requested domain which is converted to a rootDomain is not added to the result struct.

**Review by brutalgg — CHANGES_REQUESTED** [review]

File: cmd/domain.go (hunk 1 above)
Comment (DesBlock) [reply cmd/domain.go]: Updated with additional commit to resolve issue

File: cmd/domain.go (hunk 2 above)
Comment (DesBlock) [reply cmd/domain.go]: Updated with additional commit to resolve issue

**Comment by DesBlock** [comment]
Changes made, please review.

File: cmd/domain.go [hunk 3]
```diff
@@ -69,20 +69,32 @@ ScannerLoop:
...
+					result = append(result, r)
+				}
+				time.Sleep(rd)
 			} else {
-				result = append(result, r)
+				if r, e := queryDomain(rootDomain); e != nil {
+					cli.Errorln("Whois lookup error: ", e)
+				} else {
+					result = append(result, r)
+					if i != rootDomain {
```
Comment (brutalgg) [inline cmd/domain.go]: doing this check makes it unclear if the root domain was queried for or not. If the root domain is directly queried it should be included in the domains searched.

File: cmd/domain.go [hunk 4]
```diff
@@ -69,20 +69,32 @@ ScannerLoop:
...
+				} else {
+					result = append(result, r)
+				}
+				time.Sleep(rd)
 			} else {
-				result = append(result, r)
+				if r, e := queryDomain(rootDomain); e != nil {
+					cli.Errorln("Whois lookup error: ", e)
+				} else {
+					result = append(result, r)
```
Comment (brutalgg) [inline cmd/domain.go:102]: Adding the information to r and then appending r to result is cleaner then appending r and then doing the modification.

**Review by brutalgg — CHANGES_REQUESTED** [review]

**Comment by DesBlock** [comment]
Updates Completed, please review

**Review by brutalgg — APPROVED** [review]

**Pull request merged**